### ISSUE TITLE
yolo11-seg metric

### ISSUE THREAD

**Issue by Milkomeda98** [ultralytics/ultralytics#21426]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

The segmentation metrics are low, but when I visualize the results, they appear accurate. What could be causing this discrepancy?"

results_dict: {'metrics/precision(B)': 0.5177945714664283, 'metrics/recall(B)': 0.46939416134770323, 'metrics/mAP50(B)': 0.4541646141166422, 'metrics/mAP50-95(B)': 0.1459482547695784, 'metrics/precision(M)': 0.0005432635322007112, 'metrics/recall(M)': 0.0007673258693453315, 'metrics/mAP50(M)': 0.0003497671139659714, 'metrics/mAP50-95(M)': 9.730444986400078e-05, 'fitness': 0.176892441420559}

<img width="471" height="473" alt="Image" src="https://github.com/user-attachments/assets/2c0d9326-b44c-4022-bd18-e31fc2c221ea" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Milkomeda98, thank you for reaching out to Ultralytics 🚀! This is an automated response to help you get started—an Ultralytics engineer will also review your question soon.

We recommend exploring the [Docs](https://docs.ultralytics.com/), where you’ll find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, along with answers to frequently asked questions.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate further.

If your question relates to custom training or metric discrepancies, please include as much detail as possible, such as sample images, data formats, and relevant logs. Also, review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to ensure optimal outcomes.

Join the Ultralytics community for real-time discussions and support:
- [Discord](https://discord.com/invite/ultralytics) 🎧
- [Discourse](https://community.ultralytics.com/)
- [Subreddit](https://www.reddit.com/r/Ultralytics/)

## Upgrade

Please upgrade to the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

You can run YOLO in any of these verified environments (with all dependencies, including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for helping us improve Ultralytics!

**Comment by glenn-jocher** [comment]
This discrepancy typically occurs when your ground truth mask annotations don't precisely match the model's predictions at the pixel level. Even visually similar masks can have drastically different IoU scores due to small pixel misalignments.

Common causes for extremely low mask metrics (M) while having reasonable box metrics (B):

**Check your annotation format** - Ensure your ground truth masks are in the correct YOLO segmentation format (normalized polygon coordinates in .txt files) and that mask boundaries precisely match your intended segmentation areas.

**Verify dataset preparation** - Small errors in mask conversion or annotation tools can create pixel-level mismatches that severely impact [segmentation metrics](https://docs.ultralytics.com/reference/utils/metrics/#ultralytics.utils.metrics.SegmentMetrics) while still appearing visually correct.

**Review IoU thresholds** - Segmentation requires much more precise pixel-level matching than bounding box detection, making it more sensitive to annotation quality issues.

Try validating on a small subset of your data with manually verified annotations to isolate whether this is a dataset issue or a model performance issue.
